pkg/controller/registry/registry_controller.go

```diff
@@ -98,36 +96,21 @@ func (r *ReconcileRegistry) Reconcile(request reconcile.Request) (reconcile.Resu
 	originalRegistry := registry.DeepCopy()
 
 	// Fetch the Instance
-	harbor := &registriesv1alpha1.Instance{}
-	ns := types.NamespacedName{
-		Namespace: registry.Namespace,
-		Name:      registry.Spec.ParentInstance.Name,
-	}
-	err = r.client.Get(ctx, ns, harbor)
-	if errors.IsNotFound(err) {
-		registry.Status = registriesv1alpha1.RegistryStatus{
-			Name:           string(registriesv1alpha1.RegistryStatusPhaseCreating),
-			Message:        "corresponding harbor instance does not exist",
-			LastTransition: &now,
+	harbor, err := internal.FetchReadyHarborInstance(ctx, registry.Namespace, registry.Spec.ParentInstance.Name, r.client)
+	if err != nil {
+		if err == internal.ErrInstanceNotFound {
+			registry.Status = registriesv1alpha1.RegistryStatus{Name: string(registriesv1alpha1.RegistryStatusPhaseCreating)}
+			// Requeue, the instance might not have been created yet
+			return reconcile.Result{RequeueAfter: time.Second * 30}, nil
+		} else if err == internal.ErrInstanceNotReady {
+			return reconcile.Result{RequeueAfter: 120 * time.Second}, err
+		} else {
+			registry.Status = registriesv1alpha1.RegistryStatus{LastTransition: &now}
```

Comment:
Optional: There's still a bit of duplication here. This could be extracted into a shared function; since this piece of code sets the status on a concrete registry/replication/repository type, that function would instead need to operate on a generic `interface { SetStatus(registriesv1alpha1.RegistryStatus) }` type.

Reply:
I'd really like to see this, though i think a seperate PR should handle this. Created an issue to handle this: https://github.com/mittwald/harbor-operator/issues/4